packages/utils/src/npm.ts

```diff
@@ -1,157 +1,19 @@
-  }
-
-  /** Call this when the result of getNpmInfoFromCache looks potentially out-of-date. */
-  async function fetchAndCacheNpmInfo(escapedPackageName: string): Promise<NpmInfo | undefined> {
-    const info = await uncachedClient.fetchNpmInfo(escapedPackageName);
-    if (info) {
-      unroll.set(escapedPackageName, info);
-    }
-    return info;
-  }
-}
-
-export class UncachedNpmInfoClient {
-  private readonly fetcher = new Fetcher();
-
-  async fetchNpmInfo(escapedPackageName: string): Promise<NpmInfo | undefined> {
-    const raw = await this.fetchRawNpmInfo(escapedPackageName);
-    await sleep(0.01); // If we don't do this, npm resets the connection?
-    return raw === undefined ? undefined : npmInfoFromJson(raw);
-  }
-
-  async fetchRawNpmInfo(escapedPackageName: string): Promise<NpmInfoRaw | undefined> {
-    const info = (await this.fetcher.fetchJson({
-      hostname: npmRegistryHostName,
-      path: escapedPackageName,
-      retries: true,
-    })) as { readonly error: string } | NpmInfoRaw;
-    if ("error" in info) {
-      if (info.error === "Not found") {
-        return undefined;
-      }
-      throw new Error(`Error getting version at ${escapedPackageName}: ${info.error}`);
-    }
-    if (!info["dist-tags"] && !info.versions) {
-      // Unpublished
-      return undefined;
-    }
-    return info;
-  }
-
-  // See https://github.com/npm/download-counts
-  async getDownloads(packageNames: readonly string[]): Promise<readonly number[]> {
-    // NPM uses a different API if there's only a single name, so ensure there's at least 2 for every batch of 128.
-    const names = packageNames.length % 128 === 1 ? [...packageNames, "dummy"] : packageNames;
-    const nameGroups = Array.from(splitToFixedSizeGroups(names, 128)); // NPM has a limit of 128 packages at a time.
-
-    const out: number[] = [];
-    for (const nameGroup of nameGroups) {
-      const data = (await this.fetcher.fetchJson({
-        hostname: npmApi,
-        path: `/downloads/point/last-month/${nameGroup.join(",")}`,
-        retries: true,
-      })) as { readonly error: string } | { readonly [key: string]: { readonly downloads: number } };
-      if ("error" in data) {
-        throw new Error(data.error as string);
-      }
-      for (const key of Object.keys(data)) {
-        assert(
-          key === names[out.length],
-          `at index ${out.length} of ${Object.keys(data).toString()} : ${key} !== ${names[out.length]}`
-        );
-        out.push(data[key] ? data[key].downloads : 0);
-      }
-    }
-    return out;
-  }
-}
-
-function splitToFixedSizeGroups(names: readonly string[], chunkSize: number): readonly (readonly string[])[] {
-  const out: string[][] = [];
-  for (let i = 0; i < names.length; i += chunkSize) {
-    out.push(names.slice(i, i + chunkSize));
-  }
-  return out;
-}
+export const defaultCacheDir = joinPaths(
+  process.env.GITHUB_ACTIONS ? joinPaths(__dirname, "..") : os.tmpdir(),
+  "cache"
+);
```

Comment:
Moved the cache into `os.tmpdir()`, to address https://github.com/microsoft/DefinitelyTyped-tools/pull/480#issue-1266598918.

Reply:
Verified that Node can write here on the Functions host over SSH, so I’m open to giving this another try. Using pacote seems like a worthwhile simplification if we can get it to work on our infrastructure. This week is busy for me though. I saw @sandersn merged a PR in this repo today so maybe he has some time?

Reply:
I'm on DT next week so I'll look then.